### DISCUSSION TITLE
srsRAN Project 24.10

### DISCUSSION THREAD

**Discussion by brendan-mcauliffe** [srsran/srsRAN_Project#888] [Announcements]
Hey everyone, 

As some of you may have seen we released srsRAN Project 24.10 last week. This is the latest tagged release of the project and builds on the intermediate releases we have had since the last tagged release, 24.04. 

Some of the main features are as follows: 

- Added CU/DU split
- Added RAN slicing
- Added Compile time selection of split 7 and split 8
- Added E2 support in the CU

Along with these features, there have been the usual bug fixes and performance improvements, as well as a refactor and update of the configuration file. 

You can find all of the latest reference files and other information in the documentation. 

We hope you enjoy the release and as always, happy hacking!

**Comment by x0dros** [comment]
I would like to ask a couple of questions and apologies beforehand if this is the wrong thread/place to ask this. 

As I understand, currently SRS RAN is able to provide for 4x4 MIMO. When saying 4X4, are we talking about spatial streams or number of antenna elements or both? If we have a high order antenna array, can SRS RAN provide 4 spatial streams for let’s say a 32- or 64-antenna array? Can SRS handle a high number of antennas even if it can give only up to 4 spatial streams? 

Are there plans to extend SRS to support massive MIMO 5G? If yes, is there a timeline?

Thank you for your time.

**Reply by xavierarteaga** [reply]
> When saying 4X4, are we talking about spatial streams or number of antenna elements or both?

This refers to the gNodeB transmitting 4 spatial streams and the UE receiving 4 spatial streams (also called layers).

> If we have a high order antenna array, can SRS RAN provide 4 spatial streams for let’s say a 32- or 64-antenna array?

The number of spatial streams (layers) and antennas is currently limited to 4. Supporting more than 4 transmit antennas in SDR or Category A O-RU setups may not be practical.

> Can SRS handle a high number of antennas even if it can give only up to 4 spatial streams?

Yes and no. While it is possible to modify certain constants to support more than 4 antennas, there are limitations due to the TypeI Single Panel codebook PMI precoding. Additionally, supporting 32 or more antennas would require changes in NZP-CSI-RS resources, CSI reporting, codebooks, and related configurations.

To support 32 antennas or more, you would need to adjust the NZP-CSI-RS resources, CSI reporting, codebooks, and so on.

> Are there plans to extend SRS to support massive MIMO 5G? If yes, is there a timeline?

There are plans for this extension, but there is no set date for when it will be made available in the open-source project.

**Comment by rolandmartinson7890** [comment]
Thanks @brendan-mcauliffe. Very awesome development. I would like to ask if the seperation of the CU and the DU means we can now successfully have one or more computers as the DUs routing to another computer that serves as the DU?

**Comment by newt2105** [comment]
Thanks @brendan-mcauliffe and team for amazing release. I have a question, how can I do RAN slicing? Thanks!

**Reply by brendan-mcauliffe** [reply]
Slicing can be enabled via the relevant parameters in the configuration file and by configuring the core correctly. In the configuration file for the gNB, you can find the relevant parameters in the `cu_cp` under the `amf` section, and in the `cell_cfg` under the `slicing` section. 

A full tutorial is not yet available, but it is something we are working on.

**Comment by nicolasbertaud** [comment]
i have done several tests on my side  in n78 n28 n3. SRUE never complete the attachement process .  I fact even if  we provide a configuration file for srsue  with B3 or B28 o B78, srsue always behave as it is in FDD n78  4G FDD .    i always observe the following : sudo grep -iE "SSB" /tmp/ue.log
2025-08-27T19:57:22.096548 [PHY-SA ] [I] [    0] Cell search: Setting SSB configuration srate=23.04 MHz; c-freq=1842.500 MHz; ss-freq=1842.050 MHz; scs=15kHz; pattern=A; duplex=fdd;
I  came to the conclusion it is not possible to Attach srue  to gNB . Have you observed the same thing ?

**Reply by brendan-mcauliffe** [reply]
Hi @nicolasbertaud, have you followed [this guide](https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html) in the documentation?

**Reply by nicolasbertaud** [reply]
yes i have followed the documentation. I have done the test with zero MQ configuration on my side

**Reply by pgawlowicz** [reply]
try this: https://github.com/srsran/srsRAN_Project/issues/1318#issuecomment-3248314422

**Reply by pgawlowicz** [reply]
note that srsUE implements has only FDD implemented.
